core/src/main/java/com/linecorp/armeria/common/util/TimeoutMode.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.util;
+
+import com.linecorp.armeria.common.Request;
+
+/**
+ * The timeout mode.
+ */
+public enum TimeoutMode {
```

Comment:
Any thoughts on my comment here?

https://github.com/line/armeria/issues/2535#issuecomment-593757305

It prevents having adjective (from) and verb (extend) in the same enum which looks weird.

Reply:
That's a good suggestion. 👍 I know that our previous discussion does not reach a decision.
I'm still very open to this enum. 
The word meaning of `NOW` and `CURRENT` are similar. So I hesitated to use `CURRENT` a little bit.

How about adding two kinds of API.(I'm not sure this is better.)
```java
public TimeoutMode {
    NOW,
    START
}
// Schedules timeout from specific timeout mode
void setTimeout(TimeoutMode, Duration)
// Extends the previous scheduled timeout
void setTimeoutAfter(Duration)
```

Reply:
How about `FROM_OLD_TIMEOUT` or `FROM_PREV_TIMEOUT` instead of `EXTEND` then? Or we could make the two verbs, like `SET_FROM_NOW` and `SET_FROM_START`. I think the latter sounds better.

Reply:
Making all verbs with e.g. `SET_*` sounds good.

Reply:
`SET_*` is explicit, but we alread have `set` verb in method name.
```java
ctx.setTimeout(TimeoutMode.SET_FROM_NOW, Duration.ofSeconds(1))
```

Maybe we could remove `set` from method name?
```java
ctx.timeout(TimeoutMode.SET_FROM_NOW, Duration.ofSeconds(1))
ctx.timeout(TimeoutMode.SET_FROM_START, Duration.ofSeconds(1))
ctx.timeout(TimeoutMode.EXTEND, Duration.ofSeconds(1))
```

Reply:
Removing `set` from method names SGTM